src/edfi-paging-test/edfi_paging_test/api/paginated_result.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from typing import Any, Dict, List, Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from edfi_paging_test.api.request_client import RequestClient
+
+
+class PaginatedResult():
+    """
+    The PaginatedResult class is bound with information from the response of the
+    api, specifically when it returns a list of items that could be paginated.
+    Parameters
+    ----------
+    request_client : RequestClient
+        The request client.
+    page_size : int
+        The number of items per page.
+    total_count : int
+        Total number of items.
+    api_response : dict
+        The original response as a dictionary.
+    resource_name : str
+        The name used by the API for the current resource.
+    current_page : Optional[str]
+        The page that you have requested.
+    Attributes
```

Comment:
```suggestion

    Attributes
```

I've never documented attributes before. If they might be used publicly, it seems like a reasonable idea.

However, one thought: might be better to have an explicit getter, and in that getter, return a copy of the dictionary: that way external classes cannot modify the `current_page_items` collection. Similar to returning `IReadOnlyList<>` in C#.

That is something we were not very careful about in the LMS Toolkit. I only started thinking about that a week or two ago in my Data Structures class.